### DISCUSSION TITLE
Can't run test using markers along with pytest_plugins

### DISCUSSION THREAD

**Discussion by qb-nikola-popovic** [pytest-dev/pytest#10555] [Q&A]
Hello there!

This is how my code structure looks like:

```
│ api
   └───api_test.py
   └───conftest.py
│ mobile
   └───mobile_test.py
   └───conftest.py
│ web
   └───web_test.py
   └───conftest.py
```

In `api/conftest.py` and `mobile/conftest.py` I'm using pytest_plugins to invoke all the fixtures from `web/conftest.py`. Something like

```
pytest_plugins = "web.conftest"
```

Now, I want to add the `markers` to all tests I have under api/mobile/web. I'm annotating tests with:
```
@pytest.mark.api / @pytest.mark.mobile / @pytest.mark.web 
```

I've also registered these markers in `pytest.ini` file:

```
[pytest]
markers =
    api: mark app api tests
    mobile: mark mobile tests
    web: mark app web tests
```

When running tests with `pytest -m api`/`pytest -m mobile`/`pytest -m web`, I'm getting following error:

```
Defining 'pytest_plugins' in a non-top-level conftest is no longer supported:
It affects the entire test suite instead of just below the conftest as expected.
  <path/to/my/conftest>
Please move it to a top level conftest file at the rootdir:
  <path/to/my/root/dir>
For more information, visit:
  https://docs.pytest.org/en/stable/deprecations.html#pytest-plugins-in-non-top-level-conftest-files
```

When running the the tests using markers and with commenting out this part of the code `pytest_plugins = "web.conftest"` it will work (but fixtures from web/conftest.py are not getting invoked and I need them).

Anyone know how to fix this? Thank you in advance for help.

**Comment by RonnyPfannschmidt** [comment]
The reason for the error is that you would indeed make anything you put into pytest-plugins a global plugin that is applied to everything

Without more details on the setup it's unclear what to advise for

**Reply by RonnyPfannschmidt** [reply]
An "easy" solution might be to move the conftest of Web one level up as its fixtures are used everywhere

**Reply by qb-nikola-popovic** [reply]
Hi @RonnyPfannschmidt thanks for the answer!

I've tried with moving `web/conftest.py` in directory root (one level up), but same issue persists. 

In `web/conftest.py` I have fixtures for setting up selenium webdriver and generating pytest-html report.
In `api/conftest.py` I have fixtures that are creating test data, and I'm using fixtures from `web/conftest.py`.
In `mobile/conftest.py` I have fixtures for setting up appium driver and running appium server, and I'm using fixtures from `web/conftest.py` as well.

In `api/conftest.py` I have a fixture like this
```
@pytest.fixture(autouse=True, scope="session")
def create_new_onboarded_user(web_driver_init):
    ...
```
`web_driver_init` fixture is coming from `web/conftest.py`. But there are still fixtures from `web/conftest.py` that are not explicitly invoked; fixtures for generating report, like this one
```
@pytest.hookimpl(hookwrapper=True)
def pytest_runtest_makereport(item):
    ...
```

Similar setup is in `mobile/conftest.py` for using fixtures from `web/conftest.py`.

Hope it is a bit more clear now. Thanks for the answer again!
